tensorflow/examples/label_image_go/main.go

```diff
@@ -0,0 +1,214 @@
+package main
+
+/* Copyright 2015 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// A minimal but useful Go example showing how to load an Imagenet-style object
+// recognition TensorFlow model, prepare input images for it, run them through
+// the graph, and interpret the results.
+//
+// It's designed to have as few dependencies and be as clear as possible, so
+// it's more verbose than it could be in production code. The errors maagement
+// while building the Graph had been ommited in order to make the code less
+// verbose, but for production code is recomended to check the returned errors
+// on each step.
+//
+// To use it, run in a working directory with the
+// learning/brain/tutorials/label_image_go/data/ folder below it: go run
+// main.go <image_file>, you can use data/ceres_paris.jpg as image file which
+// contains the image of a Labrador and you should see the top five labels for
+// the example image output.
+//
+// The tensorflow_inception_graph.pb file included by default is created from
+// Inception.
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	tf "github.com/tensorflow/tensorflow/tensorflow/contrib/go"
+)
+
+const (
+	cInputWidth  = 299
+	cInputHeight = 299
+	cInputMean   = 128
+	cInputStd    = 128
+
+	// cInceptionGraphFile This is the path of the graph who contains the
+	// model.
+	cInceptionGraphFile = "tensorflow/examples/label_image_go/data/tensorflow_inception_graph.pb"
+	// cLabelsFile File path that contains a label per line.
+	cLabelsFile = "tensorflow/examples/label_image_go/data/imagenet_comp_graph_label_strings.txt"
+
+	// cLabelsToShow Amounth of best match labels to be returned per image.
+	cLabelsToShow = 5
+)
+
+// readTensorFromImageFile Given an image file name, read in the data, try to
+// decode it as an image, resize it to the requested size, and then scale the
+// values as desired.
+func readTensorFromImageFile(filePath string) *tf.Tensor {
+	var imageReader *tf.GraphNode
+
+	graph := tf.NewGraph()
+
+	// Load the image file path into the graph as a constant, and read its
+	// content.
+	fileNameNode, _ := graph.Constant("file_name", filePath)
+	fileReader, _ := graph.Op("ReadFile", "file_reader", []*tf.GraphNode{fileNameNode}, "", nil)
+
+	// Now try to figure out what kind of file it is and decode it.
+	if filePath[len(filePath)-4:] == ".png" {
```

Comment:
suggest
import path/filepath
if filepath.Ext(filePath) == ".png"   (does the right thing if, e.g., the file path is less than 4 characters.)